### PR TITLE
Issue #PS-2499 feat : Integration of Tracking api tracking/content/create and status api

### PR DESCRIPTION
**Pull request by suvarnakale**
…

### CODE REVIEW

File: src/components/ContentCard.tsx [hunk 1]
```diff
@@ -46,6 +55,101 @@ const ContentCard: React.FC<ContentCardProps> = ({
...
+    }
+
+    console.log('Details Object:', detailsObject);
+
+    try {
+      const contentWithTelemetryData = async () => {
+        if (userId !== undefined || userId !== '') {
+          const mimeTypesToContentTypes: { [key: string]: string } = {
+            'application/vnd.sunbird.questionset': 'quml',
+            'application/vnd.ekstep.ecml-archive': 'ecml',
```
Comment (itsvick) [inline src/components/ContentCard.tsx:119]: @suvarnakale dont hard code it here.. we are already  using this in app.constant, use it from there

File: src/components/ContentCard.tsx [hunk 2]
```diff
@@ -102,7 +206,10 @@ const ContentCard: React.FC<ContentCardProps> = ({
               : name || subTopic?.join(', ')}
           </Box>
         </Box>
-
+        <Box sx={{ width: '100%' }}>
+          <LinearProgress variant="determinate" value={status} />
```
Comment (itsvick) [inline src/components/ContentCard.tsx:210]: @suvarnakale can u attach screenshot of content card with different status

File: src/components/ContentCard.tsx [hunk 3]
```diff
@@ -46,6 +55,102 @@ const ContentCard: React.FC<ContentCardProps> = ({
...
+
+    try {
+      const contentWithTelemetryData = async () => {
+        if (userId !== undefined || userId !== '') {
+          const mimeTypesToContentTypes: { [key: string]: string } = {
+            [ContentType.ECML]: 'ecml',
+            [ContentType.QUESTION_SET]: 'quml',
+            [ContentType.HTML]: 'html',
+            [ContentType.H5P]: 'h5p',
+            [ContentType.YOUTUBE_X_VIDEO]: 'x-youtube',
```
Comment (itsvick) [inline src/components/ContentCard.tsx:122]: @suvarnakale why are u again declaring it... try something like below
const ContentTypeReverseMap = Object.fromEntries(
  Object.entries(ContentType).map(([key, value]) => [value, key])
);

**Comment by suvarnakale** [comment]
completed
inprogrss

File: public/locales/en/common.json [hunk 4]
```diff
@@ -458,7 +458,12 @@
     "PLANNED_SESSION_END_DATE_TIME_REQUIRED_ERROR": "Please provide a valid end time for the planned session.",
     "INVALID_DATE": "Invalid Date",
     "INVALID_EVENT_CONFIGURATION": "Invalid event configuration. Please check your inputs.",
-    "MEETING_URL_ERROR": "Please provide a valid meeting URL."
+    "MEETING_URL_ERROR": "Please provide a valid meeting URL.",
+    "COMPLETED": "Completed",
+    "IN_PROGRESS": "In_Progress",
+    "INPROGRESS": "In Progress",
+    "NOT_STARTED": "Not_Started",
```
Comment (itsvick) [inline public/locales/en/common.json:465]: @suvarnakale are u showing this Not_started anywhere on the UI?

File: src/services/TrackingService.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const response = await post(apiUrl, reqBody);
+    return response?.data;
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+};
+
+export const createContent = async (reqBody: ContentCreate) => {
```
Comment (itsvick) [inline src/services/TrackingService.ts:32]: @suvarnakale rename this as well its misleading... rename to createContentTracking

File: src/components/ContentCard.tsx [hunk 6]
```diff
@@ -46,6 +57,105 @@ const ContentCard: React.FC<ContentCardProps> = ({
...
+              if (content.contentId === identifier) {
+                status = content.status || '';
+                // status = parseInt(content.percentage, 10);
+                return true;
+              }
+            })
+          );
+          if (status === t('CENTER_SESSION.COMPLETED')) {
+            setProgress(100);
+            setStatusMsg(t('CENTER_SESSION.COMPLETED'));
```
Comment (itsvick) [inline src/components/ContentCard.tsx:89]: @suvarnakale use switch case here

File: public/locales/en/common.json (hunk 4 above)
Comment (suvarnakale) [reply public/locales/en/common.json:465]: no.. its coming from api response..

**Pull request merged**